app/models/register.ts

```diff
@@ -19,7 +19,7 @@ const registerSchema = new Schema<RegisterType>({
 });
 
 // Delete the associated user when deleting a register entry
-registerSchema.pre('remove', async (next) => {
+registerSchema.pre<RegisterType>('remove', async function (next) {
```

Comment:
Why change it away from an arrow function?

Reply:
Ahh, this one is funny. We reference the object using `this` below.
Mongoose will bind the object to the function's `this`. So it would not be the correct `this` for the arrow function, as the context for the `this` would then be the outermost scope (i.e. the global `this`).

Gotta love JS! :+1: